velero-plugin-for-aws/object_store.go

```diff
@@ -139,12 +139,20 @@ func (o *ObjectStore) Init(config map[string]string) error {
 
 	// AWS (not an alternate S3-compatible API) and region not
 	// explicitly specified: determine the bucket's region
+	// GetBucketRegion will attempt to get the region for a bucket using the
+	// client's configured region to determine which AWS partition to perform the query on.
+	// The request will not be signed, and will not use your AWS credentials.
 	if s3URL == "" && region == "" {
-		cfg, err := newConfigBuilder(o.log).WithTLSSettings(insecureSkipTLSVerify, caCert).Build()
+		regionCfg, err := newConfigBuilder(o.log).WithTLSSettings(insecureSkipTLSVerify, caCert).
+		// configures anonymous credentials
+		WithAnonymousCredentials().
```

Comment:
Perhaps we [don't want this if bucket isn't public](https://github.com/aws/aws-sdk-go-v2/pull/2716#pullrequestreview-2183393708).

Reply:
For future enhancement.. we can leave this for now IMO.

We [also didn't initialize credentialsFile and used anonymous when sdk was v1](https://github.com/vmware-tanzu/velero-plugin-for-aws/commit/46d9118173f8573e237ab4cc94c2e6ed83994396#diff-7ba4f8953b70f82b66f6ba15ea753c9bb847d2e01173baaef65574f60e57d86cR41).

Reply:
In any case most users already setting region in BSL config won't need this.

Reply:
Does your test case require this anonymous credential?

Reply:
It would unless we pass in credentialsFile to this config.

And unit tests don't contain credentials so we'd have to set anonymous credentials to prevent sdk from looking up default credentials in ~/.aws/credentials which in GitHub actions runner may not have.

Reply:
I'm a little confused here, for most cases the bucket would require credentials to access right?  But here for whatever buckets it just passes anonymous credential, will that cause any problem?

@kaovilai I've pinged you on slack, and let's hold this PR until we clarify, thanks.

Reply:
> General purpose buckets - Request to public buckets that grant the s3:ListBucket permission publicly do not need to be signed

https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html

The code here behind the scenes calls headbucket API.

In sdk v1 we did not pass credentials for GetBucketRegion so I'm simply restoring prior v1 behavior in v2 syntax.

I've recently added clarification in sdkv2 here
https://github.com/aws/aws-sdk-go-v2/pull/2716

Oadp has a unit test that would run using only anonymous credentials for this function specifically. 
https://github.com/openshift/oadp-operator/pull/1470/files#diff-6a76e73a3eb8fecbb69b99b27a55ed6c672b7ef2fc4d7253cbe24e66b845edca

It would fail without specifying anonymous credentials. And we don't want to commit credentials to unit tests.